include/boost/math/special_functions/nonfinite_num_facets.hpp

```diff
@@ -414,7 +414,7 @@ namespace boost {
         switch(peek_char(it, end, ct)) {
         case 'q':
         case 's':
-          if(flags_ && legacy)
+          if(flags_)
```

Comment:
Looking at the rest of the code, I assume this should be `flags_&legacy` ?  @pabristow ?

Reply:
I deny writing it! Looks reasonable, though the comments seem not very specific to me.  I can't see any reason to restrict to only legacy here. And the tests are thin.  And testing is difficult on account of the too many possible floating-point types and platforms.

Lesson to Standards Writers - make up your mind and don't duck out by making stuff implementation-dependent!